rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java

```diff
@@ -0,0 +1,50 @@
+package org.jboss.windup.rules.apps.java.decompiler;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.apache.commons.lang.StringUtils;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.WindupConfigurationModel;
+import org.jboss.windup.graph.service.WindupConfigurationService;
+import org.jboss.windup.rules.apps.java.model.JavaClassFileModel;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class DecompilerUtil
+{
+    /**
+     * Returns an appropriate output directory for the decompiled data based upon the provided {@link JavaClassFileModel}.
+     *
+     * This could be the same directory as the file itself, if the file is already in the output directory. If the .class file is referencing a file
+     * in the input directory, then this will be a classes folder underneath the output directory.
+     */
+    static File getOutputDirectoryForClass(GraphContext context, JavaClassFileModel fileModel)
```

Comment:
maybe I would state in the documentation that this is only the top-directory path (without package)